applications/ChimeraApplication/custom_processes/apply_chimera_process.cpp

```diff
@@ -0,0 +1,627 @@
+    p_new_constraint->Set(TO_ERASE);
+    mNodeIdToConstraintIdsMap[rSlaveNode.Id()].push_back(ConstraintId);
+    rMasterSlaveContainer.push_back(p_new_constraint);
+}
+
+template <int TDim>
+template <typename TVariableType>
+void ApplyChimera<TDim>::ApplyContinuityWithElement(
+    Geometry<ApplyChimera<TDim>::NodeType>& rGeometry,
+    ApplyChimera<TDim>::NodeType& rBoundaryNode,
+    Vector& rShapeFuncWeights,
+    TVariableType& rVariable,
+    unsigned int StartIndex,
+    std::vector<int>& rConstraintIdVector,
+    ApplyChimera<TDim>::MasterSlaveConstraintContainerType& rMsContainer)
+{
+    const auto& r_clone_constraint = new LinearMasterSlaveConstraint();
+    // Initialise the boundary nodes dofs to 0 at ever time steps
+    rBoundaryNode.FastGetSolutionStepValue(rVariable, 0) = 0.0;
+
+    for (IndexType i = 0; i < rGeometry.size(); i++) {
+        // Interpolation of rVariable
+        rBoundaryNode.FastGetSolutionStepValue(rVariable, 0) +=
+            rGeometry[i].GetDof(rVariable).GetSolutionStepValue(0) * rShapeFuncWeights[i];
+        AddMasterSlaveRelation(rMsContainer, *r_clone_constraint,
+                               rConstraintIdVector[StartIndex++], rGeometry[i], rVariable,
+                               rBoundaryNode, rVariable, rShapeFuncWeights[i]);
+    } // end of loop over host element nodes
+
+    // Setting the buffer 1 same buffer 0
+    rBoundaryNode.FastGetSolutionStepValue(rVariable, 1) =
+        rBoundaryNode.FastGetSolutionStepValue(rVariable, 0);
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::ReserveMemoryForConstraintContainers(
+    ModelPart& rModelPart, ApplyChimera<TDim>::MasterSlaveContainerVectorType& rContainerVector)
+{
+#pragma omp parallel
+    {
+        const ApplyChimera<TDim>::IndexType num_constraints_per_thread =
+            (rModelPart.NumberOfNodes() * 4) / omp_get_num_threads();
+#pragma omp single
+        {
+            rContainerVector.resize(omp_get_num_threads());
+            for (auto& container : rContainerVector)
+                container.reserve(num_constraints_per_thread);
+        }
+    }
+}
+
+template <int TDim>
+int ApplyChimera<TDim>::RemoveExistingConstraintsForNode(ModelPart::NodeType& rBoundaryNode)
+{
+    ConstraintIdsVectorType constrainIds_for_the_node;
+    int removed_counter = 0;
+    constrainIds_for_the_node = mNodeIdToConstraintIdsMap[rBoundaryNode.Id()];
+    for (auto const& constraint_id : constrainIds_for_the_node) {
+#pragma omp critical
+        {
+            mrMainModelPart.RemoveMasterSlaveConstraintFromAllLevels(constraint_id);
+            removed_counter++;
+        }
+    }
+    constrainIds_for_the_node.clear();
+    return removed_counter;
+}
+
+template <int TDim>
+void ApplyChimera<TDim>::AddConstraintsToModelpart(ModelPart& rModelPart,
+                                                   ApplyChimera<TDim>::MasterSlaveContainerVectorType& rContainerVector)
+{
+    ApplyChimera<TDim>::IndexType n_total_constraints = 0;
```

Comment:
```suggestion
    IndexType n_total_constraints = 0;
```
I think you don't need the class type.